sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRelation.scala

```diff
@@ -288,20 +293,28 @@ private[sql] class ParquetRelation(
     }
   }
 
+  override def unhandledFilters(filters: Array[Filter]): Array[Filter] = {
+    // The unsafe row RecordReader does not support row by row filtering so for this case
+    // it should wrap this with Spark-side filtering.
+    val enableUnsafeRowParquetReader =
+      sqlContext.getConf(SQLConf.PARQUET_UNSAFE_ROW_RECORD_READER_ENABLED.key).toBoolean
```

Comment:
Since `SQLConf.PARQUET_UNSAFE_ROW_RECORD_READER_ENABLED` is internally true by default, this `unhandledFilters` always returns `filters` as it is. So, this optimization is little meaningful to end users.
IMO if `UnsafeRowParquetRecordReader` supports row-by-row filtering, then we implement this.
Though?